File: C#/0002. Add Two Numbers.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode AddTwoNumbers(ListNode l1, ListNode l2) {
        ListNode node=new ListNode(0);
        ListNode p=node;
        int flag=0;
        while(l1!=null && l2!=null){
            if(l1.val+l2.val+flag>=10){
                p.next=new ListNode(l1.val+l2.val+flag-10);
                flag=1;
                p=p.next;
                l1=l1.next;
                l2=l2.next;
            }
            else{
                p.next=new ListNode(l1.val+l2.val+flag);
                flag=0;
                p=p.next;
                l1=l1.next;
                l2=l2.next;
            }
        }
        while(l1!=null){
            if(l1.val+flag>=10){
                p.next=new ListNode(l1.val+flag-10);
                flag=1;
                p=p.next;
                l1=l1.next;
            }
            else{
                p.next=new ListNode(l1.val+flag);
                flag=0;
                p=p.next;
                l1=l1.next;
                p.next=l1;
                break;
            }
        }
        while(l2!=null){
            if(l2.val+flag>=10){
                p.next=new ListNode(l2.val+flag-10);
                flag=1;
                p=p.next;
                l2=l2.next;
            }
            else{
                p.next=new ListNode(l2.val+flag);
                flag=0;
                p=p.next;
                l2=l2.next;
                p.next=l2;
                break;
            }
        }
        if(flag==1){
            p.next=new ListNode(1);
        }
        return node.next;
    }
}

File: archives/20190519python/0002. Add Two Numbers.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1: ListNode, l2: ListNode) -> ListNode:
        flag=0
        node=ListNode(0)
        p=node
        while l1 and l2:
            cur=l1.val+l2.val+flag
            if cur>=10:
                p.next=ListNode(cur-10)
                flag=1
            else:
                p.next=ListNode(cur)
                flag=0
            p=p.next
            l1=l1.next
            l2=l2.next
        while l1:
            cur=l1.val+flag
            if cur>=10:
                p.next=ListNode(cur-10)
                flag=1
            else:
                p.next=ListNode(cur)
                flag=0
            p=p.next
            l1=l1.next
        while l2:
            cur=l2.val+flag
            if cur>=10:
                p.next=ListNode(cur-10)
                flag=1
            else:
                p.next=ListNode(cur)
                flag=0
            p=p.next
            l2=l2.next
        if flag:
            p.next=ListNode(1)
        return node.next
